Plugins/BrickGrid/Source/BrickGrid/Private/BrickRenderComponent.cpp

```diff
@@ -328,7 +336,23 @@ UBrickRenderComponent::UBrickRenderComponent( const FObjectInitializer& Initiali
 FPrimitiveSceneProxy* UBrickRenderComponent::CreateSceneProxy()
 {
 	const double StartTime = FPlatformTime::Seconds();
-
+	TArray<EBrickClass> BrickClassByMaterial;
+	for (int32 iterator = 0; iterator < Grid->Parameters.Materials.Num(); ++iterator)
+	{
+		if (iterator == Grid->Parameters.EmptyMaterialIndex)
+			BrickClassByMaterial.Add(EBrickClass::Empty);
+		else
+		{
+			if (Grid->Parameters.Materials[iterator].SurfaceMaterial->GetBlendMode() == EBlendMode::BLEND_Translucent)
+			{
+				BrickClassByMaterial.Add(EBrickClass::Translucent);
+			}
+			else if (Grid->Parameters.Materials[iterator].SurfaceMaterial->GetBlendMode() == EBlendMode::BLEND_Opaque)
+			{
+				BrickClassByMaterial.Add(EBrickClass::Opaque);
+			}
```

Comment:
I think this could be a little bit more general by handling any blendmode other than opaque as translucent. That will handle additive, modulated, masked, etc blend modes as well as alpha-blended.